src/scheduler.ts

```diff
@@ -66,10 +67,14 @@ class DefaultTaskScheduler extends TaskScheduler {
   #maximumUsage: number;
   #onAvailableListeners: ((item: PiscinaWorker) => void)[];
 
-  constructor (maximumUsage: number) {
+  constructor (maximumUsage: number = os.availableParallelism()) {
     super(maximumUsage);
     this.#maximumUsage = maximumUsage;
     this.#onAvailableListeners = [];
+
+    if (maximumUsage > os.availableParallelism()) {
+      console.warn(`Warning: maximumUsage (${maximumUsage}) is greater than available CPU cores (${os.availableParallelism()}).`);
```

Comment:
Feel free to use https://www.npmjs.com/package/process-warning instead, has better ergonomics for these kind of situations

Reply:
Sure.